pandas/tests/tools/test_to_datetime.py

```diff
@@ -1084,7 +1084,7 @@ def test_to_datetime_array_of_dt64s(self, cache, unit):
         # numpy is either a python datetime.datetime or datetime.date
         tm.assert_index_equal(
             to_datetime(dts_with_oob, errors="ignore", cache=cache),
-            Index([dt.item() for dt in dts_with_oob]),
+            Index(dts_with_oob),
```

Comment:
can you write a small dedicated test for this.  iirc some of the non-nano stuff is liable to change this test and i dont want to accidentally get rid of coverage for this behavior